packages/widgetbook/lib/src/fields/num_input_field.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import 'field.dart';
+
+class NumInputField<T extends num> extends Field<T> {
+  NumInputField({
+    required super.name,
+    super.initialValue,
+    super.onChanged,
+    required super.codec,
```

Comment:
We can actually share the codec in the base classes, instead of providing it in all subclasses, what do you think?

```dart
super(
  codec: FieldCodec<T>(
    toParam: (value) => value.toString(),
    toValue: (param) => num.tryParse(param ?? '') as T?,
  ),
);
```